Main.java

```diff
@@ -0,0 +1,225 @@
+                team1SelectedPlayers[i] = selectedPlayer;
+                break;
+            }
+        }
+
+        System.out.println("Select 11 players for team " + team2 + ":");
+        for (int i = 0; i < 11; i++) {
+            while (true) {
+                System.out.println("Select player " + (i + 1) + ":");
+                for (int j = 0; j < playerCount[team2Index]; j++) {
+                    System.out.println(players[team2Index][j]);
+                }
+                String selectedPlayer = scanner.nextLine();
+                team2SelectedPlayers[i] = selectedPlayer;
+                break;
+            }
+        }
+
+        // Step 5: Select batting and bowling teams for first innings
+        System.out.println("Which team will bat first? " + team1 + " or " + team2 + ":");
+        String battingTeam = scanner.nextLine();
+        while (!battingTeam.equalsIgnoreCase(team1) && !battingTeam.equalsIgnoreCase(team2)) {
+            System.out.println("Invalid choice. Please select either " + team1 + " or " + team2 + ":");
+            battingTeam = scanner.nextLine();
+        }
+
+        String bowlingTeam = (battingTeam.equalsIgnoreCase(team1)) ? team2 : team1;
+        String[] battingPlayers = (battingTeam.equalsIgnoreCase(team1)) ? team1SelectedPlayers : team2SelectedPlayers;
+        String[] bowlingPlayers = (bowlingTeam.equalsIgnoreCase(team1)) ? team1SelectedPlayers : team2SelectedPlayers;
+
+        // Step 6: Get overs once for both innings
+        System.out.println("Enter the number of overs for both innings:");
+        int overs = scanner.nextInt();
+        scanner.nextLine();  // Consume newline
+
+        // Step 7: First team batting
+        int scoreTeam1 = startInnings(scanner, battingPlayers, bowlingPlayers, overs);
+
+        System.out.println("Innings complete for " + battingTeam + ". Final score: " + scoreTeam1);
+
+        // Switch roles for team 2's innings
+        String secondBattingTeam = bowlingTeam;
+        String secondBowlingTeam = battingTeam;
+        String[] secondBattingPlayers = bowlingPlayers;
+        String[] secondBowlingPlayers = battingPlayers;
+
+        System.out.println(secondBattingTeam + " needs " + (scoreTeam1 + 1) + " runs to win.");
+
+        // Step 8: Second team batting (no need to re-enter overs)
+        int scoreTeam2 = startInnings(scanner, secondBattingPlayers, secondBowlingPlayers, overs);
+
+        System.out.println("Innings complete for " + secondBattingTeam + ". Final score: " + scoreTeam2);
+
+        // Step 9: Declare winner
+        if (scoreTeam2 > scoreTeam1) {
+            System.out.println(secondBattingTeam + " wins by " + (scoreTeam2 - scoreTeam1) + " runs.");
+        } else if (scoreTeam2 == scoreTeam1) {
+            System.out.println("The match is a tie!");
+        } else {
+            System.out.println(battingTeam + " wins by " + (scoreTeam1 - scoreTeam2) + " runs.");
+        }
+
+        scanner.close();
+    }
+
```

Comment:
more than 3 parameters passed inside a method.
3 should be maximum parameter inside one method